src/org/rascalmpl/util/maven/Artifact.java

```diff
@@ -151,6 +150,11 @@ private void calculateClassPath(Scope forScope, HashSet<WithoutVersion> alreadyI
         }
     }
 
+    private Artifact createSystemArtifact(Dependency d) {
+        var path = d.getSystemPath();
+        return new Artifact(d.getCoordinate(), null, path == null ? null : Path.of(path), Collections.emptyList(), messages, null);
```

Comment:
Can we add a comment on why we don't have to check if the file exists? Since we normally a resolved artifact is one where the path is valid.

Reply:
When the file does not exist, the dependency is now kept unresolved.